lib/ansible/modules/cloud/amazon/redshift.py

```diff
@@ -243,11 +243,14 @@ def _collect_facts(resource):
         'db_name'           : resource['DBName'],
         'availability_zone' : resource['AvailabilityZone'],
         'maintenance_window': resource['PreferredMaintenanceWindow'],
+        'url'               : resource['Endpoint']['Address'],
```

Comment:
Is the `Endpoint` key always present?

Reply:
yes - my testing (with publicly_accessible yes/no or undefined) indicates the `Endpoint` key is always present.